wazuh 4.7.4 build fails with gcc 14.1.1 on Archlinux

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.7.4 | Installation | Agent | Sources | Archlinux (rolling) |

Build fails for 4.7.4 (used to work for 4.7.2) with:
```
client-agent/receiver.c: In function ‘receive_msg’:
client-agent/receiver.c:280:81: error: passing argument 4 of ‘UnmergeFiles’ from incompatible pointer type [-Wincompatible-pointer-types]
  280 |                                 if (!UnmergeFiles(file, SHAREDCFG_DIR, OS_TEXT, &ignore_list)) {
      |                                                                                 ^~~~~~~~~~~~
      |                                                                                 |
      |                                                                                 char ***
In file included from ./headers/shared.h:256,
                 from client-agent/receiver.c:11:
./headers/file_op.h:221:85: note: expected ‘const char ***’ but argument is of type ‘char ***’
  221 | int UnmergeFiles(const char *finalpath, const char *optdir, int mode, const char ***unmerged_files) __attribute__((nonnull(1)));
      |                                                                       ~~~~~~~~~~~~~~^~~~~~~~~~~~~~
client-agent/receiver.c:287:72: error: passing argument 2 of ‘cldir_ex_ignore’ from incompatible pointer type [-Wincompatible-pointer-types]
  287 |                                     if (cldir_ex_ignore(SHAREDCFG_DIR, ignore_list)) {
      |                                                                        ^~~~~~~~~~~
      |                                                                        |
      |                                                                        char **
./headers/file_op.h:346:54: note: expected ‘const char **’ but argument is of type ‘char **’
  346 | int cldir_ex_ignore(const char * name, const char ** ignore);
      |                                        ~~~~~~~~~~~~~~^~~~~~
```

GCC:
```
$ gcc --version
gcc (GCC) 14.1.1 20240507
Copyright (C) 2024 Free Software Foundation, Inc.
```
[wazuh-4.7.4.build_log.txt](https://github.com/wazuh/wazuh/files/15406334/wazuh-4.7.4.build_log.txt)

--- a/src/client-agent/receiver.c
+++ b/src/client-agent/receiver.c
@@ -284,7 +284,7 @@ int receive_msg()
                                     send_msg(msg_output, -1);
                                 }
                                 else {
-                                    if (cldir_ex_ignore(SHAREDCFG_DIR, ignore_list)) {
+                                    if (cldir_ex_ignore(SHAREDCFG_DIR, (const char **)ignore_list)) {
                                         mwarn("Could not clean up shared directory.");
                                     }
                                     clear_merged_hash_cache();

--- a/src/headers/file_op.h
+++ b/src/headers/file_op.h
@@ -230,7 +230,7 @@ int MergeAppendFile(FILE *finalfp, const char *files, int path_offset) __attribu
  * @param mode Indicates if the merged file must be readed as a binary file  or not. Use `#OS_TEXT`, `#OS_BINARY`.
  * @return 1 if the file was unmerged, 0 on error.
  */
-int UnmergeFiles(const char *finalpath, const char *optdir, int mode, const char ***unmerged_files) __attribute__((nonnull(1)));
+int UnmergeFiles(const char *finalpath, const char *optdir, int mode, char ***unmerged_files) __attribute__((nonnull(1)));
 
 
 /**

--- a/src/shared/file_op.c
+++ b/src/shared/file_op.c
@@ -623,7 +623,7 @@ void DeleteState() {
 }
 
 
-int UnmergeFiles(const char *finalpath, const char *optdir, int mode, const char ***unmerged_files)
+int UnmergeFiles(const char *finalpath, const char *optdir, int mode, char ***unmerged_files)
 {
     int ret = 1;
     int state_ok;

